transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollRecvByteAllocatorMessageHandle.java

```diff
@@ -31,7 +31,9 @@ public boolean continueReading() {
          * If edgeTriggered is used we need to read all bytes/messages as we are not notified again otherwise. For
          * packet oriented descriptors must read until we get a EAGAIN
          * (see Q9 in <a href="http://man7.org/linux/man-pages/man7/epoll.7.html">epoll man</a>).
+         *
+         * If RDHUP has been received we must read until we get a read error.
          */
-        return isEdgeTriggered() ? true : super.continueReading();
+        return isRdHup() || isEdgeTriggered() ? true : super.continueReading();
```

Comment:
Maybe flip the isRdHup() || isEdgeTriggered() as I I think isEdgeTriggered() is true by default anyway and so by default we not need to evaluate both things